apps/mark/frontend/src/app_setup_errors.test.tsx

```diff
@@ -200,52 +192,43 @@ describe('Displays setup warning messages and errors screens', () => {
         reload={jest.fn()}
       />
     );
-    const getByTextWithMarkup = withMarkup(screen.getByText);
-
-    // Let the initial hardware detection run.
-    await advanceTimersAndPromises();
+    const findByTextWithMarkup = withMarkup(screen.findByText);
 
     // Start on Insert Card screen
-    screen.getByText(insertCardScreenText);
+    await screen.findByText(insertCardScreenText);
 
     // Remove charger and reduce battery level slightly
     act(() => {
       hardware.setBatteryDischarging(true);
       hardware.setBatteryLevel(0.6);
     });
-    await advanceTimersAndPromises(BATTERY_POLLING_INTERVAL / 1000);
```

Comment:
Does removing these significantly slow down the test overall? I think it's good to use fake timers and advance them when we have any sort of significant polling delay.

Reply:
Nope, it has no impact on the speed of the test. Just confirmed. This test file uses fake timers and the `waitFor` and `findBy...` APIs are also advancing fake timers between each of their checks.